nifi-nar-bundles/nifi-hubspot-bundle/nifi-hubspot-processors/src/main/java/org/apache/nifi/processors/hubspot/GetHubSpot.java

```diff
@@ -0,0 +1,257 @@
+            CRM_ENDPOINT,
+            LIMIT,
+            WEB_CLIENT_SERVICE_PROVIDER
+    ));
+
+    private static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+            REL_SUCCESS,
+            REL_FAILURE
+    )));
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        webClientServiceProvider = context.getProperty(WEB_CLIENT_SERVICE_PROVIDER).asControllerService(WebClientServiceProvider.class);
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTY_DESCRIPTORS;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final String accessToken = context.getProperty(ACCESS_TOKEN).getValue();
+        final String endpoint = context.getProperty(CRM_ENDPOINT).getValue();
+
+        final StateMap state = getStateMap(context);
+        final URI uri = createUri(context, state);
+
+        final HttpResponseEntity response = getHttpResponseEntity(accessToken, uri);
+        final AtomicInteger objectCountHolder = new AtomicInteger();
+
+        FlowFile flowFile = session.create();
+        flowFile = session.putAttribute(flowFile, "statusCode", String.valueOf(response.statusCode()));
+
+        flowFile = session.write(flowFile, out -> {
+
+            try (JsonParser jsonParser = JSON_FACTORY.createParser(response.body());
+                 final JsonGenerator jsonGenerator = JSON_FACTORY.createGenerator(out, JsonEncoding.UTF8)) {
+                while (jsonParser.nextToken() != null) {
+                    if (jsonParser.getCurrentToken() == JsonToken.FIELD_NAME && jsonParser.getCurrentName().equals("results")) {
+                        jsonParser.nextToken();
+                        jsonGenerator.copyCurrentStructure(jsonParser);
+                        objectCountHolder.incrementAndGet();
+                    }
+                    String fieldName = jsonParser.getCurrentName();
+                    if (CURSOR_PARAMETER.equals(fieldName)) {
+                        jsonParser.nextToken();
+                        Map<String, String> newStateMap = new HashMap<>(state.toMap());
+                        newStateMap.put(endpoint, jsonParser.getText());
+                        updateState(context, newStateMap);
+                        break;
+                    }
+                }
+            }
+        });
+        if (response.statusCode() >= 400) {
+            if (response.statusCode() == 429) {
+                context.yield();
+                throw new ProcessException("Rate limit exceeded, yielding for 10 seconds before retrying request.");
+            } else {
+                getLogger().warn("HTTP [{}] client error occurred at endpoint [{}]", response.statusCode(), endpoint);
```

Comment:
Recommend a minor adjustment to remove the `[]` characters around the status code since it will always be a number, and also adjusting the wording to include the complete URI.
```suggestion
                getLogger().warn("HTTP {} error for requested URI [{}]", response.statusCode(), uri);
```

Reply:
I usually avoid logging URIs for security but in this case the base URI is public. only the headers contain sensitive information. Thanks.